packages/lib/modules/vebal/vebal-navigation.ts

```diff
@@ -0,0 +1,31 @@
+import { useSearchParams } from 'next/navigation'
+
+export const fromManage = 'manage'
+export const fromVote = 'vote'
+
+// Used to determine the redirect path after a vebal lock action
+type VeBalSourcePage = 'manage' | 'vote'
+
+export function getVeBalManagePath(
+  veBalAction: 'lock' | 'unlock' | 'extend',
+  sourcePage: VeBalSourcePage
+) {
+  return `/vebal/manage/${veBalAction}?from=${sourcePage}`
+}
+
+export function useVeBalRedirectPath() {
+  const searchParams = useSearchParams()
+  const from = searchParams.get('from')
+
+  let redirectPath = '/vebal/'
```

Comment:
Should the default page be `/vebal` or the previous one even if no search param has been set? I was thinking about how can we do this more generalized (like using a context that saves the last visited page or something similar to https://github.com/vercel/next.js/discussions/36723#discussioncomment-2698954)